metals/src/main/scala/scala/meta/internal/metals/OnTypeFormattingProvider.scala

```diff
@@ -60,16 +78,23 @@ final class OnTypeFormattingProvider(
       if (pipeInScope(pos, sourceText)) {
         val tokens =
           Input.VirtualFile(source.toString(), sourceText).tokenize.toOption
-        tokens.flatMap { tokens =>
-          tokens.collectFirst {
-            case token: Constant.String
-                if inToken(pos, token) &&
-                  isMultilineString(sourceText, token) =>
-              new TextEdit(range, indent(sourceText, pos) + "|")
+        tokens.flatMap { tokens: Tokens =>
+          val tokenIndex = tokens.indexWhere {
```

Comment:
Just a though, no need to act on it.

it would be nice to have the tokens as something akin to the iterator:
```scala
tokens.skipUntil{
 case token: Constant.String => (...)
 case _ =>  false
}
```

the `isFinishedByStripMargin` implementation would be much nicer

Reply:
This would imply that Tokens are mutable underneath or I would need to create a mutable structure over it :/ I prefer to keep any mutable state local and explicit.

Reply:
And I needed to change it even more to make it work with interpolated strings